src/core-services/account/mutations/setUserPermissions.js

```diff
@@ -0,0 +1,80 @@
+import SimpleSchema from "simpl-schema";
+import ReactionError from "@reactioncommerce/reaction-error";
+
+const inputSchema = new SimpleSchema({
+  "accountId": String,
+  "userId": String,
+  "groups": {
+    type: Array, // groupIds that user belongs to
+    optional: true,
+    defaultValue: []
+  },
+  "groups.$": {
+    type: String
+  }
+});
+
+/**
+ * @name accounts/setUserPermissions
+ * @memberof Mutations/Accounts
+ * @summary Adds a group to an account (user) group. This addition to an account  group effectively
+ * adds permissions to account (user)
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Necessary input for mutation. See SimpleSchema.
+ * @param {Object} input.groups - groups to append to
+ * @param {String} input.accountId - optional decoded ID of account on which entry should be updated, for admins
+ * @returns {Promise<Object>} with updated account
+ */
+export default async function setUserPermissions(context, input) {
+  const itemsToValidate = { accountId: context.accountId, userId: context.userId, groups: input.groups };
+  inputSchema.validate(itemsToValidate);
```

Comment:
This should be validating the input object that's going to be used in the `update` operation. This is currently validating the existence of the `accountId` and `userId` of the currently signed-in user, not the user that you're trying to update the groups for.

Reply:
@mikemurray ... I've updated the mutation function to update the account of the `accountId` supplied in the input. I've also updated the validations to validate the input for the `accountId` and  `groups` supplied in the input